dist/tools/flake8/check.sh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+#
+# Copyright (C) 2017 Alexandre Abadie <alexandre.abadie@inria.fr>
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+if tput colors &> /dev/null && [ $(tput colors) -ge 8 ]; then
+    CERROR="\e[1;31m"
+    CWARNING="\e[1;33m"
+    CRESET="\e[0m"
+else
+    CERROR=
+    CWARNING=
+    CRESET=
+fi
+
+. ${RIOTBASE:+${RIOTBASE}/}dist/tools/ci/changed_files.sh
+
+FILES=$(FILEREGEX='*.py$' changed_files)
+
+if [ -z "${FILES}" ]
+then
+    exit 0
+fi
+
+# Allow 119 characters width per line to simplify test output parsing
+ERRORS=$(flake8 --max-line-length=119 ${FILES})
+
+if [ -n "${ERRORS}" ]
+then
+    echo -e "${CWARNING}There are style issues in the following Python scripts:${CRESET}"
```

Comment:
AFAIR "echo -e" causes trouble on some platforms (OSX?).

Reply:
printf is better, more portable